daemon/policy.go

```diff
@@ -114,7 +122,7 @@ func (d *Daemon) UpdatePolicyEnforcement(e *endpoint.Endpoint) bool {
 			d.conf.Opts.Set(endpoint.OptionPolicy, false)
 			return false
 		}
-	} else if d.conf.EnablePolicy == endpoint.DefaultEnforcement && d.conf.IsK8sEnabled() {
+	} else if d.conf.EnablePolicy == endpoint.DefaultEnforcement && d.conf.IsK8sEnabled() && e != nil {
```

Comment:
Not sure if intencional or not, but be aware the `e != nil` is only checked if k8s mode is enabled.